src/transformers/generation/utils.py

```diff
@@ -1221,6 +1208,55 @@ def _prepare_generation_config(
 
         return generation_config, model_kwargs
 
+    def _prepare_special_tokens(
+        self, generation_config: GenerationConfig, kwargs_has_attention_mask: Optional[bool] = None
```

Comment:
`kwargs_has_attention_mask` is an optional argument so we can use this function in tests, to prepare special tokens.